### ISSUE TITLE
Parser should ignore files with hashbangs

### ISSUE THREAD

**Issue by silverwind** [vuejs/vue-eslint-parser#161]
Parsing a file like 

```js
#!/usr/bin/env node
console.info('hello world');
```

results in a parser error with this parser:

```
Error while parsing /file.js

Line 1, column 2: Unexpected character '!'
`parseForESLint` from parser `/file.js` is invalid and will just be ignored
```

eslint's own parser handles this by ignoring files starting with hashbang:

https://github.com/eslint/eslint/blob/b93af98b3c417225a027cabc964c38e779adb945/lib/linter/linter.js#L779

**Comment by ota-meshi** [comment]
I don't think it makes sense to use hashbangs in vue files.

**Comment by silverwind** [comment]
It doesn't and I'd actually like to restrict the parser to just `.vue` files, but they way [eslint-plugin-vue](https://github.com/vuejs/eslint-plugin-vue) configures the eslint parser, this parser is used for all files, not just `.vue` and I haven't found a way yet on how to restrict this parser.

**Comment by ota-meshi** [comment]
Please provide a minimal repository to reproduce the problem.

**Comment by silverwind** [comment]
I already worked around the issue:

```yaml
overrides:
  - files: ["**/*.vue"]
    parser: vue-eslint-parser
```

Arguably, this is how `plugin:vue` should probably do it as well, not replace the parser for all files, but only those needed ones.

**Comment by silverwind** [comment]
If you want to reproduce: 

```
git clone --depth 1 https://github.com/go-gitea/gitea && cd gitea && npm install && eslint build/*.js
```

This will reproduce until https://github.com/go-gitea/gitea/pull/20113 is merged.

**Comment by ota-meshi** [comment]
You seem to be aware of it already.

https://github.com/import-js/eslint-plugin-import/issues/1369

**Comment by silverwind** [comment]
Yeah, https://github.com/import-js/eslint-plugin-import/pull/2431 will fix it in `eslint-module-utils`, but I don't see that module in the dependencies of `vue-eslint-parser`, so I have my doubts whether that PR will fix the issue.
